client/src/js/services/journal/JournalGrouping.js

```diff
@@ -0,0 +1,86 @@
+angular.module('bhima.services')
+.service('JournalGroupingService', JournalGroupingService);
+
+JournalGroupingService.$inject = [];
+
+/**
+ * Posting Journal Grouping Service 
+ *
+ * This service is responsible for setting the grouping configuration for the 
+ * client side posting journal module. It also provides a number of helper 
+ * methods that can be used to provide custom transaction grouping. 
+ */
+function JournalGroupingService() { 
+  var service = this;
+  
+  // Variable used to track and share the current grids API object
+  var gridApi;
+ 
+  /**
+   * This method enables group level selections when the posting jounral is grouped
+   * according to a column. It first checks to ensure the row that changed is a row 
```

Comment:
Minor typo: jounral  :arrow_right: journal.

Slightly more serious typo: "It first checks to ensure the row that changed is a row" should probably read "It first checks to ensure that the row that changed is a __header__ row and then selects the row's children."